syntax = "proto3";

service Authorization { // Server
  rpc Register (RegisterRequest) returns (RegisterReply) {}
  rpc Login (LoginRequest) returns (LoginReply) {}
}

// Status codes for registration
enum RegisterCodeResult
{
  RCR_undefined = 0;
  RCR_ok = 1;
  RCR_already_exist = 2;
  RCR_bad_credentials = 3;
}

// Status codes for login
enum LoginCodeResult
{
  LCR_undefined = 0;
  LCR_ok = 1;
  LCR_unknown_user = 2;
  LCR_bad_credentials = 3;
}

// Request for registration
message RegisterRequest {
  string user_name = 1;
  string user_passwd = 2;
}

// Server response to registration request
message RegisterReply {
  RegisterCodeResult code = 1;
  string reason = 2;
}

// Request for login
message LoginRequest {
  string user_name = 1;
  string user_passwd = 2;
}

// Server response to login request
message LoginReply {
  LoginCodeResult code = 1;
  string reason = 2;
  string token = 3;
}
